lib/endpoints/class-wp-rest-terms-controller.php

```diff
@@ -193,6 +192,20 @@ public function create_item( $request ) {
 			$args['slug'] = $request['slug'];
 		}
 
+		if ( isset( $request['parent'] ) ) {
+			if ( ! is_taxonomy_hierarchical( $this->taxonomy ) ) {
+				return new WP_Error( 'rest_taxonomy_not_hierarchical', __( "Can not set term parent, taxonomy is not hierarchical." ), array( 'status' => 400 ) );
+			}
```

Comment:
`parent` isn't present in the schema, so we don't need to worry about the taxonomy check.